syntax = "proto3";

package bank.account;

//거래 로그 구조체
message TransactionLog{
	uint32 logId = 1;
	string accNum = 2;
	string userId = 3;
	uint32 transaction_type = 4;
	uint64 trade_amount = 5;
	uint64 remaining_balance = 6;
	string created_at = 7;
}

//계좌 기본 정보
message AccountInfo{
	string accNum = 1;
	string userId = 2;
	uint32 account_type = 3;
	bool status = 4;
	string created_at = 5;
}

// CreateFixedDepositAccount
message CreateFixedDepositAccountRequest {
	string userId = 1;
	double interestRate = 2;
	uint64 monthly_payment = 3;
	string duedate = 4;
}

message CreateFixedDepositAccountResponse {
	bool status = 1;
	string message = 2;

	string accNum = 3;
}

// CreateNormalSavingsAccount
message CreateNormalSavingsAccountRequest {
	string userId = 1;
	double interestRate = 2;
}

message CreateNormalSavingsAccountResponse {
	bool status = 1;
	string message = 2;

	string accNum = 3;
}

// ListTransactionLogsByAccNum
message ListTransactionLogsRequest {
	string accNum = 1;
}

message ListTransactionLogsResponse {
	bool status = 1;
	string message = 2;

	repeated TransactionLog logs = 3;
}

// ListAccountsByUserId
message ListAccountsByUserIdRequest {
	string userId = 1;
}

message ListAccountsByUserIdResponse {
	bool status = 1;
	string message = 2;

	repeated AccountInfo accounts = 3;
}

//Transfer
message TransferRequest{
	string userId = 1;
	string srcAccNum = 2;
	string destAccNum = 3;
	uint64 trade_amount = 4;
}

message TransferResponse{
	bool status = 1;
	string message = 2;

	TransactionLog log = 3;
}
